simulator/resourceapplier/resourceapplier_test.go

```diff
@@ -0,0 +1,521 @@
+			if err != nil {
+				t.Fatalf("failed to get pod when comparing: %v", err)
+			}
+			var gotPod corev1.Pod
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(got.UnstructuredContent(), &gotPod)
+			if err != nil {
+				t.Fatalf("failed to convert got unstructured to pod: %v", err)
+			}
+
+			if diff := cmp.Diff(*tt.podAfterUpdate, gotPod); diff != "" {
+				t.Errorf("updatePods() mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestResourceApplier_deletePods(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name    string
+		pod     *corev1.Pod
+		wantErr bool
+	}{
+		{
+			name: "delete a Pod",
+			pod: &corev1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pod-1",
+					Namespace: "default",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "container-1",
+							Image: "image-1",
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			client, mapper := prepare()
+			service := New(client, mapper, Options{})
+
+			p, err := runtime.DefaultUnstructuredConverter.ToUnstructured(tt.pod)
+			if err != nil {
+				t.Fatalf("failed to convert pod to unstructured: %v", err)
+			}
+			unstructedPod := &unstructured.Unstructured{Object: p}
+			err = service.Create(context.Background(), unstructedPod)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("deletePods() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			err = service.Delete(context.Background(), unstructedPod)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("deletePods() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			_, err = client.Resource(corev1.Resource("pods").WithVersion("v1")).Namespace("default").Get(context.Background(), tt.pod.Name, metav1.GetOptions{})
+			if err == nil {
+				t.Fatalf("pod should be deleted but it still exists")
+			}
+
+			if !errors.IsNotFound(err) && !tt.wantErr {
+				t.Fatalf("failed to check if pod is deleted: %v", err)
+			}
+		})
+	}
+}
+
+func prepare() (*dynamicFake.FakeDynamicClient, meta.RESTMapper) {
+	s := runtime.NewScheme()
+	v1.AddToScheme(s)
+	scheduling.AddToScheme(s)
+	storage.AddToScheme(s)
+	client := dynamicFake.NewSimpleDynamicClient(s)
+	resources := []*restmapper.APIGroupResources{
+		{
+			Group: metav1.APIGroup{
+				Versions: []metav1.GroupVersionForDiscovery{
+					{Version: "v1"},
+				},
+			},
+			VersionedResources: map[string][]metav1.APIResource{
+				"v1": {
+					{Name: "pods", Namespaced: true, Kind: "Pod"},
+				},
+			},
+		},
+		{
+			Group: metav1.APIGroup{
+				Versions: []metav1.GroupVersionForDiscovery{
+					{Version: "v1"},
+				},
+			},
+			VersionedResources: map[string][]metav1.APIResource{
+				"v1": {
+					{Name: "nodes", Namespaced: true, Kind: "Node"},
```

Comment:
```suggestion
					{Name: "nodes", Namespaced: false, Kind: "Node"},
```

Nodes are actually cluster-scope.
Having both cluster-scope and namespaced resources in the test would improve completeness.

Reply:
I added `createNodes` for the case of cluster-scope resources.